src/herbivoresAndCarnivores.js

```diff
@@ -1,16 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden === true ? this.hidden = false : this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
-}
+  bite(prey) {
+    if (prey.hidden === false
+      && prey instanceof Herbivore) {
+      prey.health -= 50;
+
+      for (let i = 0; i < Animal.alive.length; i++) {
+        if (Animal.alive[i].health === 0) {
+          Animal.alive.splice(i, 1);
+        }
```

Comment:
remove loop, just check the health of current animal and then filter it if it less than 0

Reply:
fix this